virtualhosts/etc/e-smith/templates/etc/opt/rh/rh-php72/php-fpm.d/000-virtualhost.conf/10Base

```diff
@@ -0,0 +1,73 @@
+{
+    use esmith::ConfigDB;
+    my $vdb = esmith::ConfigDB->open_ro('vhosts') || die("Can't open vhosts db");
+
+    foreach my $vhost ($vdb->get_all_by_prop('type'=>'vhost')) {
+      
+        my $VhostName = $vhost->key;
+        my $status = $vhost->prop('status') || 'disabled';
+        my $PhpRhVersion = $vhost->prop('PhpRhVersion') ||' default';
+
+        #skip if vhost status is not enabled
+        if ($status ne 'enabled' ){
+          next;
+        }
+        #skip if vhost php version is not php72
+        if ($PhpRhVersion ne 'php72' ){
+          next;
+        }
+
+        my $PhpCustomSettings = $vhost->prop('PhpCustomSettings') || 'disabled';
+        my $ServerNames = $vhost->prop('ServerNames') || '';
+        $ServerNames =~ s/,/ /g;
+
+        $OUT .= qq(
+;
+; vhosts $ServerNames
+;
+
+[$VhostName]
+listen = /var/run/rh-$PhpRhVersion-php-fpm/$VhostName-$PhpRhVersion.sock
+
+;Logs
+php_admin_value[error_log] = /var/opt/rh/rh-$PhpRhVersion/log/php-fpm/error.log
+php_admin_flag[log_errors] = on
+
+pm = dynamic
+pm.max_children = 50
+pm.start_servers = 5
+pm.min_spare_servers = 5
+pm.max_spare_servers = 35
+
+user = apache;
+group = apache
+listen.owner = root
+listen.group = apache
+listen.mode = 0660
+
+; Set data paths to directories owned by process user
+php_value[session.save_handler] = files
+php_value[session.save_path]    = /var/opt/rh/rh-$PhpRhVersion/lib/php/session
+php_value[soap.wsdl_cache_dir]  = /var/opt/rh/rh-$PhpRhVersion/lib/php/wsdlcache
+
+);
+
+        if ($PhpCustomSettings eq 'enabled'){
+
+            my $AllowUrlfOpen = $vdb->get_prop("$VhostName",'AllowUrlfOpen') || 'enabled';
+            my $MemoryLimit = $vdb->get_prop("$VhostName",'MemoryLimit') || '0';
+            my $UpMaxFileSize = $vdb->get_prop("$VhostName",'UploadMaxFilesize') || '0';
+            my $PostMaxSize = $vdb->get_prop("$VhostName",'PostMaxSize') || '0';
+            my $MaxExecTime = $vdb->get_prop("$VhostName",'MaxExecutionTime') || '0';
+            my $MaxFileUploads = $vdb->get_prop("$VhostName",'MaxFileUploads') || 'disabled';
+            
+            $OUT .= "php_admin_flag[allow_url_fopen] = off\n" if ($AllowUrlfOpen eq 'disabled');
+            $OUT .= "php_admin_value[memory_limit] = $MemoryLimit"."M\n" if ($MemoryLimit ne '0');
+            $OUT .= "php_admin_value[upload_max_filesize] = $UpMaxFileSize"."M\n" if ($UpMaxFileSize ne '0');
+            $OUT .= "php_admin_value[post_max_size] = $PostMaxSize"."M\n" if ($PostMaxSize ne '0');
+            $OUT .= "php_admin_value[max_execution_time] = $MaxExecTime\n" if ($MaxExecTime ne 'disabled');
+            $OUT .= "php_admin_value[max_input_time] =  $MaxExecTime\n" if ($MaxExecTime ne 'disabled');
+            $OUT .= "php_admin_value[max_file_uploads] = $MaxFileUploads\n" if ($MaxFileUploads ne 'disabled');
+        }
```

Comment:
Why not adding here an `else` branch?

If custom settings are disabled we set the defaults values from the `php` key settings in ConfigDB:

```text
[root@vm5 test]# config show php
php=configuration
    DateTimezone=UTC
    ExposePhp=
    MaxExecutionTime=0
    MemoryLimit=256
    PostMaxSize=8
    ShortOpenTag=
    UploadMaxFilesize=2
```

However there are some hardcoded default settings to take into account...

https://github.com/NethServer/nethserver-httpd/blob/96770d692d0213d5e7099481ed4a0f390bb1bd51/ui/src/views/Settings.vue#L107-L112

Reply:
not sure to understand well, if PhpCustomSettings is disabled, then the values comes from the php.ini of the php pool IIRC, why do you want to enable others values

Reply:
yep confirmed, the values come from `/etc/opt/rh/rh-php72/php.ini` if not set by our UI default are

```
max_execution_time	30
memory_limit	128M
post_max_size	8M
upload_max_filesize	2M
```

do you prefer others values from our php settings (only differences is `MaxExecutionTime`)

Reply:
Yes, if custom values are not enabled I'd expect values from the Settings page are applied. There is no other information provided by the UI about the PHP configurations.

Reply:
ok, lets code